docs/intro-react.md

```diff
@@ -0,0 +1,476 @@
+
+You can declare a component’s state by calling `useState` inside its function like this:
+
+`const [currentValue, setNewValue] = useState("Initial value.");`
+
+You can use `useState` to track any kind of data, including numbers:
+
+`const [quantity, pickQuantity] = useState(0);`
+
+Calling `useState` does two things:
+
+- it creates a “state variable” with an initial value
+- it creates a function to set that state variable’s value
+
+Let’s break `[a, b] = useState(c)` down:
+
+- `a` is the “state variable”—it refers to the current value of the component’s state
+- `b` is the name of the function you use to update the state variable
+- `c` is the value you initialize the state variable with
+
+```SnackPlayer name=State
+import React, { useState } from 'react';
+import { Text, TextInput, View } from 'react-native';
+
+function Cat() {
+
+  const [name, setName] = useState("What's my name again?");
+
+  return (
+    <View>
+      <Text>Hello, I am...{name}!</Text>
+      <TextInput
+        style={{
+          height: 40,
+          borderColor: 'gray',
+          borderWidth: 1
+        }}
+        onChangeText={
+          (text) => setName(text)
+        }
+      />
+    </View>
+  );
+}
+
+export default function Cafe() {
+  return (
+    <>
+      <Cat />
+      <Cat />
+    </>
+  );
+}
+```
+
+> See the `<>` and `</>` above? These bits of JSX are [fragments](https://reactjs.org/docs/fragments.html). Adjacent JSX elements must be wrapped in an enclosing tag. Fragments let you do that without nesting an extra, unnecessary wrapping element like `View`.
+
+First, you will want to import `useState` from React like so:
+
+```jsx
+import React, {useState} from 'react';
+```
+
+In this example, inside your `Cat` function, `useState` creates a `name` state variable:
+
+```jsx
+const [name, setName] = useState("What's my name again?");
+```
+
+Then you add `onChangeText` to the `<TextInput>`. This fires every time the `<TextInput>`‘s value changes (every time a user adds a character of text).
+
+```jsx
+onChangeText={(text) => setName(text)}
+```
+
+This uses the `setName` you defined with the `useState` Hook above to update your state variable `defaultText` every time `onChangeText` fires. You might set state when you have new data from the server, or from user input, or even from a timer.
+
+> You might’ve noticed that although `name` is a [const](https://developer.mozilla.org/Web/JavaScript/Reference/Statements/const), it is seemingly reassignable! What is happening is that every time a user inputs a letter into `<TextInput>`, `setName()` fires to update the state const `name`. When a state-setting function like `setName` is called, its component will re-render, running its function again. In this case `Cat` will re-render its `<Cat>`—discarding the old state const `name` in the process!
```

Comment:
This is good to call out.

Reply:
Thanks! I made an engineer friend who came to visit read these docs and he tripped on this one!